src/LetsEncrypt/Internal/CertificateSelector.cs

```diff
@@ -48,8 +48,10 @@ public void ClearChallengeCert(string domainName)
             _challengeCerts.TryRemove(domainName, out _);
         }
 
-        private static void AddWithDomainName(ConcurrentDictionary<string, X509Certificate2> certs, string domainName, X509Certificate2 certificate)
+        private void AddWithDomainName(ConcurrentDictionary<string, X509Certificate2> certs, string domainName, X509Certificate2 certificate)
         {
+            PreloadIntermediateCertificates(certificate);
```

Comment:
I initially added it to the ConcurentDictionary update lambda, but chain loading could and probably will make several blocking calls. I believe CD update will lock the key internally, so the update should be as fast as possible.